test-framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -571,6 +564,19 @@ private static String groupName(ThreadGroup threadGroup) {
     }
 
     /**
+     * Returns size random values
+     */
+    public static <T> List<T> randomSubsetOf(int size, Collection<T> values) {
```

Comment:
The existing `ESTestCase#randomSubsetOf(int, Object...)` should just delegate to this new method here.